Azure/azure-cli
# [Compute] Fix #20133: `az vm create`: Fix `--data-disk-delete-option` not working when no `--attach-data-disks` are provided

Issue: #20133 

**Description**

Fix #20133 data-disk-delete-option was not working when no attach-data-disks are provided with it.

**Testing Guide**

`az vm create --data-disk-delete-option Delete --resource-group {rg} --name {vm-name} --image UbuntuLTS`
After VM deletion, the data disk created should be deleted.

**History Notes**


[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
